tools/souper-check.cpp

```diff
@@ -134,87 +137,96 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
+        if (std::error_code EC = S->powerTwo(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                             PowTwo, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known powerOfTwo from souper: "
+                       << convertToStr(PowTwo) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferNonZero) {
-      bool NonZero;
-      if (std::error_code EC = S->nonZero(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                          NonZero, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
-        llvm::outs() << "known nonZero from souper: "
-                     << convertToStr(NonZero) << "\n";
-        ++Success;
+      if (InferNonZero) {
+        bool NonZero;
+        if (std::error_code EC = S->nonZero(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                            NonZero, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known nonZero from souper: "
+                       << convertToStr(NonZero) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferSignBits) {
-      unsigned SignBits;
-      if (std::error_code EC = S->signBits(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                           SignBits, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
-        llvm::outs() << "known signBits from souper: "
-                     << std::to_string(SignBits) << "\n";
+      if (InferSignBits) {
+        unsigned SignBits;
+        if (std::error_code EC = S->signBits(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                             SignBits, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known signBits from souper: "
+                       << std::to_string(SignBits) << "\n";
+          ++Success;
+        }
+      }
+      if (InferRange) {
+        unsigned W = Rep.Mapping.LHS->Width;
+        llvm::ConstantRange Range = S->constantRange(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS, IC);
+
+        llvm::outs() << "known range from souper: " << "[" << Range.getLower()
+                     << "," << Range.getUpper() << ")" << "\n";
         ++Success;
       }
-    } else if (InferRange) {
-      unsigned W = Rep.Mapping.LHS->Width;
-      llvm::ConstantRange Range = S->constantRange(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS, IC);
-
-      llvm::outs() << "known range from souper: " << "[" << Range.getLower()
-                   << "," << Range.getUpper() << ")" << "\n";
-      ++Success;
-    } else if (InferRHS || ReInferRHS) {
+    }
+    else if (InferRHS || ReInferRHS) {
```

Comment:
please put else on the same line as its closing curly brace

Reply:
also, if the "if" has curly braces, then the "else" must also

Reply:
fixed:)

Reply:
wait, it doesn't look fixed
I am seeing this:
<img width="382" alt="Screen Shot 2019-09-15 at 7 40 41 PM" src="https://user-images.githubusercontent.com/732536/64930737-b88ecd00-d7f0-11e9-9ec1-2e5646fbd31e.png">
and saying that it should look more like this:
```
} else if ...
```